cmake/modules/boards.cmake

```diff
@@ -195,6 +195,24 @@ endif()
 
 if(NOT "${ret_board}" STREQUAL "")
   string(STRIP "${ret_board}" ret_board)
+  string(FIND "${ret_board}" "\n" idx REVERSE)
+  if(idx GREATER -1)
+    while(TRUE)
+      math(EXPR start "${idx} + 1")
+      string(SUBSTRING "${ret_board}" ${start} -1 line)
+      string(SUBSTRING "${ret_board}" 0 ${idx} ret_board)
+
+      cmake_parse_arguments(LIST_BOARD "" "DIR" "" ${line})
+      set(board_dirs "${board_dirs}\n${LIST_BOARD_DIR}")
+
+      if(idx EQUAL -1)
+        break()
+      endif()
+      string(FIND "${ret_board}" "\n" idx REVERSE)
+    endwhile()
+    message(FATAL_ERROR "Multiple boards named '${BOARD}' found in:${board_dirs}")
```

Comment:
I would prefer we detect this at the `list_boards.py` level, especially considering that in future we need to also support this: https://github.com/zephyrproject-rtos/zephyr/issues/69548

as well as ensuring that other commands, like `west boards` detect such errors as well.

However, as immediate fix, then this proposal is accepted and can be cleaned up when implementing https://github.com/zephyrproject-rtos/zephyr/issues/69548.

Reply:
> I would prefer we detect this at the `list_boards.py` level

Agreed, and I'd like to get around to this as soon as I can.

I've already started work on refactoring the Python scripts, which would make this easier to detect at Python level, and would include performance improvements among other things.